### PR TITLE
fall update

### PR DESCRIPTION
**Pull request by jelmerj**
My attempt at merging some changes as well :)

### CODE REVIEW

**Comment by Azurency** [comment]
Thank for your contribution 👍 I'll be looking at your files and test it tomorrow (I hope) !

File: Assets/UI/PartialScreens/citystates.lua [hunk 1]
```diff
@@ -430,6 +430,14 @@ function Close()
         UI.PlaySound("CityStates_Panel_Close");
     end
 
+    local localPlayerID = Game.GetLocalPlayer();
+    if (localPlayerID ~= -1) then
+    	local localPlayer = Players[localPlayerID];
+    	if (localPlayer ~= nil and localPlayer:GetInfluence() ~= nil and not localPlayer:GetInfluence():IsGivingTokensConsidered()) then
+    		localPlayer:GetInfluence():SetGivingTokensConsidered(true);
+    	end
+    end
```
Comment (Azurency) [inline Assets/UI/PartialScreens/citystates.lua]: The indentation of this block is not correct, we need 2 spaces (here it's a mix between spaces and tabs)

File: Assets/UI/PartialScreens/citystates.lua [hunk 2]
```diff
@@ -849,7 +849,9 @@ function AddCityStateRow( kCityState:table )
 
   textureOffsetX, textureOffsetY, textureSheet, tooltip = GetRelationshipPipAtlasPieces( kCityState );
   kInst.DiplomacyPip:SetTexture( textureOffsetX, textureOffsetY, textureSheet );
-  kInst.DiplomacyPip:SetToolTipString(tooltip);
+  if GameCapabilities.HasCapability("CAPABILITY_MILITARY") then
+  	kInst.DiplomacyPip:SetToolTipString(tooltip);
```
Comment (Azurency) [inline Assets/UI/PartialScreens/citystates.lua]: indentation

File: Assets/UI/PartialScreens/citystates.lua [hunk 3]
```diff
@@ -1116,7 +1121,9 @@ function ViewCityState( iPlayer:number )
 
       textureOffsetX, textureOffsetY, textureSheet, tooltip = GetRelationshipPipAtlasPieces( kCityState );
       kInst.DiplomacyPip:SetTexture( textureOffsetX, textureOffsetY, textureSheet );
-      kInst.DiplomacyPip:SetToolTipString(tooltip);
+      if GameCapabilities.HasCapability("CAPABILITY_MILITARY") then
+      			kInst.DiplomacyPip:SetToolTipString(tooltip);
```
Comment (Azurency) [inline Assets/UI/PartialScreens/citystates.lua]: indentation

File: Assets/UI/PartialScreens/citystates.lua [hunk 4]
```diff
@@ -1143,39 +1150,51 @@ function ViewCityState( iPlayer:number )
...
+    Controls.LevyMilitaryButton:SetHide(false);
+    Controls.LevyMilitaryButton:SetDisabled(not kCityState.CanLevyMilitary);
+    if kCityState.HasLevyActive and kCityState.IsLocalPlayerSuzerain then
+    	local levyTooltip = Locale.Lookup("LOC_CITY_STATES_MILITARY_ALREADY_LEVIED");
+    	Controls.LevyMilitaryButton:SetToolTipString(levyTooltip);
+    else
+    	local levyTooltip = Locale.Lookup("LOC_CITY_STATES_LEVY_MILITARY_DETAILS", kCityState.LevyMilitaryCost, kCityState.LevyMilitaryTurnLimit);
+    	Controls.LevyMilitaryButton:SetToolTipString(levyTooltip);
+    end 
+    Controls.LevyMilitaryButton:RegisterCallback( Mouse.eLClick, function() OnLevyMilitary( kCityState ); end );
```
Comment (Azurency) [inline Assets/UI/PartialScreens/citystates.lua]: from line 1153 to 1193, the indentation is not correct

File: Assets/UI/PartialScreens/citystates.lua [hunk 5]
```diff
@@ -1285,15 +1304,12 @@ function ViewCityState( iPlayer:number )
...
-      if (pLocalPlayer ~= nil) then
-        local pPlayerConfig :table = PlayerConfigurations[iOtherPlayer];
-        if (localPlayerID == iOtherPlayer) then
-          civName = Locale.Lookup(pPlayerConfig:GetPlayerName()) .. " (" .. Locale.Lookup("LOC_CITY_STATES_YOU") .. ")";
-        elseif (pLocalPlayer:GetDiplomacy():HasMet(iOtherPlayer)) then
-          civName = pPlayerConfig:GetPlayerName();
-        end
+      local pPlayerConfig :table = PlayerConfigurations[iOtherPlayer];
+      if (localPlayerID == iOtherPlayer) then
+      	civName = Locale.Lookup(pPlayerConfig:GetPlayerName()) .. " (" .. Locale.Lookup("LOC_CITY_STATES_YOU") .. ")";
```
Comment (Azurency) [inline Assets/UI/PartialScreens/citystates.lua]: indentation is slightly off here (missing 1 space)

File: Assets/UI/PartialScreens/citystates.lua [hunk 6]
```diff
@@ -1480,26 +1498,7 @@ function GetData()
...
-        cityStateType = "TRADE";
-      elseif (leader == "LEADER_MINOR_CIV_CULTURAL" or leaderInfo.InheritFrom == "LEADER_MINOR_CIV_CULTURAL") then
-        cityStateType = "CULTURE";
-      elseif (leader == "LEADER_MINOR_CIV_MILITARISTIC" or leaderInfo.InheritFrom == "LEADER_MINOR_CIV_MILITARISTIC") then
-        cityStateType = "MILITARISTIC";
-      elseif (leader == "LEADER_MINOR_CIV_INDUSTRIAL" or leaderInfo.InheritFrom == "LEADER_MINOR_CIV_INDUSTRIAL") then
-        cityStateType = "INDUSTRIAL";
-      end
-
+			local cityStateType	:string = GetCityStateType( iPlayer );
```
Comment (Azurency) [inline Assets/UI/PartialScreens/citystates.lua]: indentation

File: Assets/UI/PartialScreens/citystates.lua [hunk 7]
```diff
@@ -1604,6 +1603,31 @@ function GetData()
...
+	elseif (leader == "LEADER_MINOR_CIV_CULTURAL" or leaderInfo.InheritFrom == "LEADER_MINOR_CIV_CULTURAL") then
+		cityStateType = "CULTURE";
+	elseif (leader == "LEADER_MINOR_CIV_MILITARISTIC" or leaderInfo.InheritFrom == "LEADER_MINOR_CIV_MILITARISTIC") then
+		cityStateType = "MILITARISTIC";
+	elseif (leader == "LEADER_MINOR_CIV_INDUSTRIAL" or leaderInfo.InheritFrom == "LEADER_MINOR_CIV_INDUSTRIAL") then
+		cityStateType = "INDUSTRIAL";
+	end
+
+	return cityStateType;
+end
```
Comment (Azurency) [inline Assets/UI/PartialScreens/citystates.lua]: Indentation of this function

File: Assets/UI/PartialScreens/citystates.lua [hunk 8]
```diff
@@ -1655,11 +1679,13 @@ end
...
 function OnCityLiberated(playerID:number, cityID:number)
-  local localPlayerID = Game.GetLocalPlayer();
-  if (localPlayerID == -1) then
-    return;
+  if not ContextPtr:IsHidden() then
+  	local localPlayerID = Game.GetLocalPlayer();
+  	if (localPlayerID == -1) then
+  		return;
+  	end
+  	Refresh();
```
Comment (Azurency) [inline Assets/UI/PartialScreens/citystates.lua]: indentation of this `if` here also

File: Assets/UI/PartialScreens/citystates.lua [hunk 9]
```diff
@@ -1685,29 +1713,35 @@ function OnDiplomacyMakePeace(firstPlayerID, secondPlayerID)
       m_kEnvoyChanges = {}; -- Zero out any pending envoy choices
     end
   end
-  Refresh();
+  if not ContextPtr:IsHidden() then
+  	Refresh();
```
Comment (Azurency) [inline Assets/UI/PartialScreens/citystates.lua]: indentation

File: Assets/UI/PartialScreens/citystates.lua [hunk 10]
```diff
@@ -1685,29 +1713,35 @@ function OnDiplomacyMakePeace(firstPlayerID, secondPlayerID)
...
 function OnInfluenceChanged()
-  local localPlayerID = Game.GetLocalPlayer();
-  if (localPlayerID == -1) then
-    return;
+  if not ContextPtr:IsHidden() then
+  	local localPlayerID = Game.GetLocalPlayer();
+  	if (localPlayerID == -1) then
+  		return;
+  	end
+  	Refresh();
```
Comment (Azurency) [inline Assets/UI/PartialScreens/citystates.lua]: indentation of this `if`

File: Assets/UI/PartialScreens/citystates.lua [hunk 11]
```diff
@@ -1685,29 +1713,35 @@ function OnDiplomacyMakePeace(firstPlayerID, secondPlayerID)
...
 function OnInfluenceGiven()
-  local localPlayerID = Game.GetLocalPlayer();
-  if (localPlayerID == -1) then
-    return;
+  if not ContextPtr:IsHidden() then
+  	local localPlayerID = Game.GetLocalPlayer();
+  	if (localPlayerID == -1) then
+  		return;
+  	end
+  	Refresh();
```
Comment (Azurency) [inline Assets/UI/PartialScreens/citystates.lua]: same, indentation

File: Assets/UI/PartialScreens/citystates.lua [hunk 12]
```diff
@@ -1723,11 +1757,13 @@ end
...
 function OnQuestChanged()
-  local localPlayerID = Game.GetLocalPlayer();
-  if (localPlayerID == -1) then
-    return;
+  if not ContextPtr:IsHidden() then
+  	local localPlayerID = Game.GetLocalPlayer();
+  	if (localPlayerID == -1) then
+  		return;
+  	end
+  	Refresh();
```
Comment (Azurency) [inline Assets/UI/PartialScreens/citystates.lua]: same

**Comment by jelmerj** [comment]
Thanks for the review, should be fixed now!

**Comment by Azurency** [comment]
Thank you very much for this contribution, you're now merge into CQUI 🎉 ! 
You can follow the remaining progress in the fall issue and maybe be one of the tester if you can next week before we release an updated version on steam !

**Pull request merged**